joint_state_controller/src/joint_state_controller.cpp

```diff
@@ -53,8 +53,14 @@ namespace joint_state_controller
       return false;
     }
 
+    std::string pub_namespace;
+    if (controller_nh.getParam("namespace", pub_namespace))
+    {
+        ROS_DEBUG("Publishing joint states under namespace '%s'", pub_namespace.c_str());
+    }
+
     // realtime publisher
-    realtime_pub_.reset(new realtime_tools::RealtimePublisher<sensor_msgs::JointState>(root_nh, "joint_states", 4));
+    realtime_pub_.reset(new realtime_tools::RealtimePublisher<sensor_msgs::JointState>(root_nh, pub_namespace + "/joint_states", 4));
```

Comment:
This would publish to `/joint_states` per default, but it should be `joint_states` (best practice: never hard code absolute namespaces!).

Reply:
To fix this issue, you should intoduce a `topic` parameter (default: `joint_states`) and pass this to the `RealtimePublisher` as-is.

Reply:
I had tried adding a remap argument to the launch file that launched the controller spawner, but that didn't seem to work (and I assumed that was the result of the remapping not being propagated through the controller spawner). What's the right syntax?

Agreed on your second point. I coded it that way based on my wrong assumption above that namespacing/topic remapping didn't apply.

Reply:
I believe the following syntax _should_ do the remapping that I want, but this hasn't had the desired effect:

```xml
<remap to="/gazebo/joint_states" from="joint_states"/>
<rosparam file="$(find myrobot_gazebo)/controller/joint_state_controller.yaml" command="load"/>
<node name="myrobot_joint_state_controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn joint_state_controller" respawn="false" output="screen"/>
```
I've also tried reversing the to and from arguments in case I was misinterpreting the docs.

Reply:
The `spawner` is just a "remote control", you have to apply the remap to the node running the `controller_manager`, so `gazebo` in your case.
Instead of remapping, you could as well set the [`robotNamespace` for the Gazebo plugin](https://github.com/ros-simulation/gazebo_ros_demos/blob/kinetic-devel/rrbot_description/urdf/rrbot.xml#L18).

Reply:
Ah, that certainly makes sense now that you say it. 

Well I obviously should've gone to ROS answers first, but thanks for your help! I'll close this PR.